File: agents/android-agent/app/src/main/java/com/livedashboard/agent/ReportPayload.kt
package com.livedashboard.agent

data class ReportPayload(
    val appId: String,
    val windowTitle: String,
    val isForeground: Boolean,
    val timestamp: String,
    val extra: ExtraInfo
)

data class ExtraInfo(
    val batteryPercent: Int? = null,
    val batteryCharging: Boolean? = null
)

File: agents/ios/LiveDashboardAgent/LiveDashboardAgent/Models/ReportPayload.swift
import Foundation

struct ReportPayload {
    let appId: String
    let windowTitle: String
    let isForeground: Bool
    let timestamp: String
    let extra: ExtraInfo

    func toDict() -> [String: Any] {
        var extraDict: [String: Any] = [:]
        if let bp = extra.batteryPercent { extraDict["battery_percent"] = bp }
        if let bc = extra.batteryCharging { extraDict["battery_charging"] = bc }
        if let so = extra.screenOn { extraDict["screen_on"] = so }

        return [
            "app_id": appId,
            "window_title": windowTitle,
            "is_foreground": isForeground ? "1" : "0",
            "timestamp": timestamp,
            "extra": extraDict
        ]
    }
}
